lib/SILGen/SILGenFunction.cpp

```diff
@@ -696,6 +696,11 @@ void SILGenFunction::emitGeneratorFunction(SILDeclRef function, VarDecl *var) {
     if (originalProperty->isPropertyWrapperInitializedWithInitialValue()) {
       interfaceType = originalProperty->getValueInterfaceType();
       varType = originalProperty->getType();
+      if (originalProperty->
+          isInnermostPropertyWrapperInitUsesEscapingAutoClosure()) {
+        interfaceType = FunctionType::get({}, interfaceType);
+        varType = FunctionType::get({}, varType);
```

Comment:
Another place where we could use that new function on `VarDecl`, them `varType` is `mapTypeIntoContext(the-wrappedValue-init-type)`.